### PR TITLE
expand /api/reviews endpoint to have queries

### PR DESCRIPTION
**Pull request by akashdeep931**

### CODE REVIEW

File: __tests__/integration.test.js [hunk 1]
```diff
@@ -114,6 +114,59 @@ describe("GET /api/reviews", () => {
...
+          expect(review).toHaveProperty("category", expect.any(String));
+          expect(review).toHaveProperty("review_img_url", expect.any(String));
+          expect(review).toHaveProperty("created_at", expect.any(String));
+          expect(review).toHaveProperty("votes", expect.any(Number));
+          expect(review).toHaveProperty("designer", expect.any(String));
+          expect(review).toHaveProperty("comment_count", expect.any(String));
+        });
+      });
+  });
+  it("400: should return an error when given an incorrect query", () => {
```
Comment (johnhutcheon) [inline __tests__/integration.test.js:139]: This should be a 404

File: __tests__/integration.test.js [hunk 2]
```diff
@@ -114,6 +114,59 @@ describe("GET /api/reviews", () => {
...
+        const { reviews } = body;
+
+        expect(reviews).toHaveLength(13);
+        expect(reviews).toBeSortedBy("owner", { descending: false });
+      });
+  });
+  it("400: should return an error when given incorrect queries", () => {
+    return request(app)
+      .get("/api/reviews?sort_by=names&order=first")
+      .expect(400)
```
Comment (johnhutcheon) [inline __tests__/integration.test.js:163]: Could we separate this into two tests please? One for sort_by and one for order, both 400s 👍

**Review by johnhutcheon — CHANGES_REQUESTED** [review]
Thanks for making these changes, you've just missed the final one! If you could just add this one please?

- Status 200, valid category query, but has no articles responds with an empty array of articles, e.g. ?category=paper.

File: __tests__/integration.test.js [hunk 3]
```diff
@@ -114,6 +114,59 @@ describe("GET /api/reviews", () => {
...
+      .then(({ body }) => {
+        const { msg } = body;
+
+        expect(msg).toBe("Bad Request!");
+      });
+  });
```
[the remark this answers is not in the document]
Comment (akashdeep931) [reply __tests__/integration.test.js:179]: Hey John, what does that mean at all, I mean, if a category query does not exists in the reviews table I will automatically return 404 Not Found! I am not sure if I am understanding that correctly.

**Pull request merged**